equinix/resource_fabric_cloud_router.go

```diff
@@ -307,16 +308,144 @@ func resourceFabricCloudRouterRead(ctx context.Context, d *schema.ResourceData,
+			stringValues := converters.IfArrToStringArr(values.([]interface{}))
+			filterExpression.SetValues(stringValues)
+		}
+
+		// If the parent has any contents then all the children schema properties will be included in the map even
+		// if they aren't given a value. Still need to check for empty string for the value because of this.
+		if orGroup, ok := filterMap["or"]; ok && orGroup.(bool) {
+			orValues := cloudRouterOrFilter.GetOr()
+			orValues = append(orValues, filterExpression)
+			if len(orValues) > 3 {
+				return fabricv4.CloudRouterFilters{}, fmt.Errorf("too many OR group filters passed. Passed %d but can only have a maximum of 3", len(orValues))
+			}
+			cloudRouterOrFilter.SetOr(orValues)
+		} else {
+			cloudRouterFilter.CloudRouterSimpleExpression = &filterExpression
+			cloudRouterFiltersList = append(cloudRouterFiltersList, cloudRouterFilter)
+		}
+	}
+
+	if orGroupHasValues := cloudRouterOrFilter.GetOr(); len(orGroupHasValues) > 0 {
+		cloudRouterFilter := fabricv4.CloudRouterFilter{}
+		cloudRouterFilter.CloudRouterOrFilter = &cloudRouterOrFilter
+		cloudRouterFiltersList = append(cloudRouterFiltersList, cloudRouterFilter)
+	}
+
+	cloudRouterFilters := fabricv4.CloudRouterFilters{}
+	cloudRouterFilters.SetAnd(cloudRouterFiltersList)
+
+	if len(cloudRouterFilters.GetAnd()) > 8 {
+		return fabricv4.CloudRouterFilters{}, fmt.Errorf("too many filters are applied to the data source. The maximum is 8 and %d were provided. Please reduce your filter count to 8", len(cloudRouterFilters.GetAnd()))
+	}
+
+	return cloudRouterFilters, nil
+}
+
+func cloudRouterPaginationTerraformToGo(pagination []interface{}) fabricv4.PaginationRequest {
+	if pagination == nil || len(pagination) == 0 {
+		return fabricv4.PaginationRequest{}
+	}
+	paginationRequest := fabricv4.PaginationRequest{}
+	for _, page := range pagination {
+		pageMap := page.(map[string]interface{})
+		if offset, ok := pageMap["offset"]; ok {
+			paginationRequest.SetOffset(int32(offset.(int)))
+		}
+		if limit, ok := pageMap["limit"]; ok {
+			paginationRequest.SetLimit(int32(limit.(int)))
+		}
+	}
+
+	return paginationRequest
+}
+
+func cloudRouterSortTerraformToGo(sort []interface{}) []fabricv4.CloudRouterSortCriteria {
+	if sort == nil || len(sort) == 0 {
+		return []fabricv4.CloudRouterSortCriteria{}
+	}
+	sortCriteria := make([]fabricv4.CloudRouterSortCriteria, len(sort))
+	for index, item := range sort {
+		sortItem := fabricv4.CloudRouterSortCriteria{}
+		pageMap := item.(map[string]interface{})
+		if direction, ok := pageMap["direction"]; ok {
+			sortItem.SetDirection(fabricv4.CloudRouterSortDirection(direction.(string)))
+		}
+		if property, ok := pageMap["property"]; ok {
+			sortItem.SetProperty(fabricv4.CloudRouterSortBy(property.(string)))
+		}
+		sortCriteria[index] = sortItem
+	}
+	return sortCriteria
+}
+
+func resourceFabricCloudRoutersSearch(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
It seems like all of this new code in the resource file is only used by the new multi-result data source; in that case this code could live alongside the rest of the new data source code in, e.g., `data_source_fabric_cloud_routers.go`

Reply:
Agreed. The remaining changes in this file do relate to code that was already there. A refactor was made so that we can set the data response easily for a GET by UUID or a POST with search Parameters that has the same response schema as elements inside of the data list.